diff --git a/lib/module/AnimatedLineGraph.js b/lib/module/AnimatedLineGraph.js
index f08d7f4aff6f8006a1c41ae7e0860552ed13f0ed..e6ac8462f69ae1ef8d11adcc8f40f3fd4b069700 100644
--- a/lib/module/AnimatedLineGraph.js
+++ b/lib/module/AnimatedLineGraph.js
@@ -77,14 +77,14 @@ export function AnimatedLineGraph({
     setHeight(Math.round(layout.height));
   }, []);
   const straightLine = useMemo(() => {
-    const path = Skia.Path.Make();
-    path.moveTo(0, height / 2);
+    const pathBuilder = Skia.PathBuilder.Make();
+    pathBuilder.moveTo(0, height / 2);
     for (let i = 0; i < width - 1; i += 2) {
       const x = i;
       const y = height / 2;
-      path.cubicTo(x, y, x, y, x, y);
+      pathBuilder.cubicTo(x, y, x, y, x, y);
     }
-    return path;
+    return pathBuilder.build();
   }, [height, width]);
   const paths = useSharedValue({});
   const gradientPaths = useSharedValue({});
diff --git a/lib/module/CreateGraphPath.js b/lib/module/CreateGraphPath.js
index 85c30c5e38a2bd5f05ea11fe5ff93ce4b41f7714..e93b8fac5d0c7eb1b4298a699b203f7764505538 100644
--- a/lib/module/CreateGraphPath.js
+++ b/lib/module/CreateGraphPath.js
@@ -49,13 +49,13 @@ function createGraphPathBase({
   canvasWidth: width,
   shouldFillGradient
 }) {
-  const path = Skia.Path.Make();
+  const pathBuilder = Skia.PathBuilder.Make();
 
   // Canvas width substracted by the horizontal padding => Actual drawing width
   const drawingWidth = width - 2 * horizontalPadding;
   // Canvas height substracted by the vertical padding => Actual drawing height
   const drawingHeight = height - 2 * verticalPadding;
-  if (graphData[0] == null) return path;
+  if (graphData[0] == null) return pathBuilder.build();
   const points = [];
   const startX = getXInRange(drawingWidth, graphData[0].date, range.x) + horizontalPadding;
   const endX = getXInRange(drawingWidth, graphData[graphData.length - 1].date, range.x) + horizontalPadding;
@@ -91,7 +91,7 @@ function createGraphPathBase({
     const point = points[i];
 
     // first point needs to start the path
-    if (i === 0) path.moveTo(point.x, point.y);
+    if (i === 0) pathBuilder.moveTo(point.x, point.y);
     const prev = points[i - 1];
     const prevPrev = points[i - 2];
     if (prev == null) continue;
@@ -103,15 +103,14 @@ function createGraphPathBase({
     const cp2y = (p0.y + 2 * p1.y) / 3;
     const cp3x = (p0.x + 4 * p1.x + point.x) / 6;
     const cp3y = (p0.y + 4 * p1.y + point.y) / 6;
-    path.cubicTo(cp1x, cp1y, cp2x, cp2y, cp3x, cp3y);
+    pathBuilder.cubicTo(cp1x, cp1y, cp2x, cp2y, cp3x, cp3y);
     if (i === points.length - 1) {
-      path.cubicTo(point.x, point.y, point.x, point.y, point.x, point.y);
+      pathBuilder.cubicTo(point.x, point.y, point.x, point.y, point.x, point.y);
     }
   }
+  const path = pathBuilder.build();
   if (!shouldFillGradient) return path;
-  const gradientPath = path.copy();
-  gradientPath.lineTo(endX, height + verticalPadding);
-  gradientPath.lineTo(0 + horizontalPadding, height + verticalPadding);
+  const gradientPath = Skia.PathBuilder.MakeFromPath(path).lineTo(endX, height + verticalPadding).lineTo(0 + horizontalPadding, height + verticalPadding).build();
   return {
     path: path,
     gradientPath: gradientPath
diff --git a/src/AnimatedLineGraph.tsx b/src/AnimatedLineGraph.tsx
index 6b1241741b610c0a80306d292af79017f188f0ee..38b5fed6992df78a44c284aad20bd426228c71a1 100644
--- a/src/AnimatedLineGraph.tsx
+++ b/src/AnimatedLineGraph.tsx
@@ -133,15 +133,15 @@ export function AnimatedLineGraph({
   );
 
   const straightLine = useMemo(() => {
-    const path = Skia.Path.Make();
-    path.moveTo(0, height / 2);
+    const pathBuilder = Skia.PathBuilder.Make();
+    pathBuilder.moveTo(0, height / 2);
     for (let i = 0; i < width - 1; i += 2) {
       const x = i;
       const y = height / 2;
-      path.cubicTo(x, y, x, y, x, y);
+      pathBuilder.cubicTo(x, y, x, y, x, y);
     }
 
-    return path;
+    return pathBuilder.build();
   }, [height, width]);
 
   const paths = useSharedValue<{ from?: SkPath; to?: SkPath }>({});
diff --git a/src/CreateGraphPath.ts b/src/CreateGraphPath.ts
index 5676695e91e8b86cd64df18096ecf57556a16788..ab4ef436f8915b5cc93183074afbf32b637e1983 100644
--- a/src/CreateGraphPath.ts
+++ b/src/CreateGraphPath.ts
@@ -144,14 +144,14 @@ function createGraphPathBase({
 }: GraphPathConfigWithGradient | GraphPathConfigWithoutGradient):
   | SkPath
   | GraphPathWithGradient {
-  const path = Skia.Path.Make();
+  const pathBuilder = Skia.PathBuilder.Make();
 
   // Canvas width substracted by the horizontal padding => Actual drawing width
   const drawingWidth = width - 2 * horizontalPadding;
   // Canvas height substracted by the vertical padding => Actual drawing height
   const drawingHeight = height - 2 * verticalPadding;
 
-  if (graphData[0] == null) return path;
+  if (graphData[0] == null) return pathBuilder.build();
 
   const points: SkPoint[] = [];
 
@@ -211,7 +211,7 @@ function createGraphPathBase({
     const point = points[i]!;
 
     // first point needs to start the path
-    if (i === 0) path.moveTo(point.x, point.y);
+    if (i === 0) pathBuilder.moveTo(point.x, point.y);
 
     const prev = points[i - 1];
     const prevPrev = points[i - 2];
@@ -227,19 +227,21 @@ function createGraphPathBase({
     const cp3x = (p0.x + 4 * p1.x + point.x) / 6;
     const cp3y = (p0.y + 4 * p1.y + point.y) / 6;
 
-    path.cubicTo(cp1x, cp1y, cp2x, cp2y, cp3x, cp3y);
+    pathBuilder.cubicTo(cp1x, cp1y, cp2x, cp2y, cp3x, cp3y);
 
     if (i === points.length - 1) {
-      path.cubicTo(point.x, point.y, point.x, point.y, point.x, point.y);
+      pathBuilder.cubicTo(point.x, point.y, point.x, point.y, point.x, point.y);
     }
   }
 
-  if (!shouldFillGradient) return path;
+  const path = pathBuilder.build();
 
-  const gradientPath = path.copy();
+  if (!shouldFillGradient) return path;
 
-  gradientPath.lineTo(endX, height + verticalPadding);
-  gradientPath.lineTo(0 + horizontalPadding, height + verticalPadding);
+  const gradientPath = Skia.PathBuilder.MakeFromPath(path)
+    .lineTo(endX, height + verticalPadding)
+    .lineTo(0 + horizontalPadding, height + verticalPadding)
+    .build();
 
   return { path: path, gradientPath: gradientPath };
 }
